From 645cefecafc3b40f4cb91675703335bfdc8f8b91 Mon Sep 17 00:00:00 2001
From: Allen Winter <winter@kde.org>
Date: Sat, 7 Jan 2012 13:54:24 -0500
Subject: [PATCH] in c4_String::c4_String(), only call memset() if the number
 of bytes to set is greater than 0.

BUG: 290688
MERGE: 4.8
(cherry picked from commit cd7e237dde2d0a81a49d6f5e73636870ff3f7fb1)
---
 .../plugins/mk4storage/metakit/src/string.cpp      |   26 +++++++++++---------
 1 file changed, 14 insertions(+), 12 deletions(-)

diff --git a/akregator/plugins/mk4storage/metakit/src/string.cpp b/akregator/plugins/mk4storage/metakit/src/string.cpp
index 81adb33..f727335 100644
--- a/akregator/plugins/mk4storage/metakit/src/string.cpp
+++ b/akregator/plugins/mk4storage/metakit/src/string.cpp
@@ -120,7 +120,9 @@ c4_String::c4_String(char ch, int n /* =1 */) {
   _value = new unsigned char[n + 3];
 
   _value[0] = 1; // see Init() member
-  memset(_value + 2, ch, n);
+  if ( n > 0 ) {
+    memset(_value + 2, ch, n);
+  }
   _value[1] = (unsigned char)(n <= 255 ? n : 255);
   _value[n + 2] = 0;
 }
-- 
1.7.10

